### PR TITLE
Update UG and DG

### PR DESCRIPTION
**Pull request by jiaweiteo**
Finalise UG
Finalise DG
Change version in Main
Update README

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1164,4 +1338,54 @@ testers are expected to do more *exploratory* testing.
...
+The `Storage` was extended to store more classes of _wanderlust_. `activity`, `accommodation`, and `friend` objects have to be organised and
+stored in a `travelplan`, and all these data will be placed in a JSON file in an organised manner. Essentially, the JSON file will contain 
+data of various `travelplan`s and a `wishlist`, and each `travelplan` will contain more data regarding `activity`, `accommodation`, and `friend`, 
+with each `travelplan` object storing its own attributes as well.
+
+Furthermore, complex attributes such as `wanderlustdate` and `wanderlustdatetime` have to be stored in the JSON file in an approriate format.
+Significant time and effort were put into discussing and planning the best way to represent each new attribute in the JSON file.
+
+### Ui
+The `Ui` component took us the most amount of time. We have to redesign the entire AB3 Ui to fit _wanderlust_ desigin.
```
Comment (lyeyixian) [inline docs/DeveloperGuide.md:1379]: Typo here. "design"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1164,4 +1338,54 @@ testers are expected to do more *exploratory* testing.
...
+
+### Logic
+The complexity of `Logic` was more advanced than AB3. _wanderlust_'s `Logic` supports
+many more commands compared to AB3, such as `sort`, `goto`, `copy`, `move`, `show` commands. Existing commands from AB3 were also
+enhanced to handle the complex `Model` of _wanderlust_.
+
+There was significant effort placed in creating the logic for _wanderlust. The `Logic` of _wanderlust is responsible for 
+at least 3x the `Logic` of AB3. It has to account for the logic of `travelplan`, `wishlist`, `activity`, `accommodation` and `friend` whilst 
+AB3 handles `Person`. Furthermore, additional logic was implemented to account for the navigation of directories and travel plans.
+For example, `ObservableDirectory` class was created to keep track of what `travelplan` the user is currently on so that the `Ui` will be
```
Comment (lyeyixian) [inline docs/DeveloperGuide.md:1364]: "directory" the user is currently on?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1164,4 +1338,54 @@ testers are expected to do more *exploratory* testing.
...
+
+Furthermore, complex attributes such as `wanderlustdate` and `wanderlustdatetime` have to be stored in the JSON file in an approriate format.
+Significant time and effort were put into discussing and planning the best way to represent each new attribute in the JSON file.
+
+### Ui
+The `Ui` component took us the most amount of time. We have to redesign the entire AB3 Ui to fit _wanderlust_ desigin. 
+Alot of effort was placed in planning the structure of the `Ui` so that our user interface will be user-friendly and appealing.
+Furthermore, additional logic of navigating between travel plans and switching between the activity/accommodation/friend tabs was implemented 
+so that users can use the application smoothly. We also spent a considerable amount of time in studying javafxml so that we are able to implement an 
+entirely new UI with icons and navigation options, making it a more intractive experience for the users as compared to the single page view of AB3's Ui.
```
Comment (lyeyixian) [inline docs/DeveloperGuide.md:1383]: Typo "interactive"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1164,4 +1338,54 @@ testers are expected to do more *exploratory* testing.
...
+
+### Overall
+All in all, this project was a challenging one for the _wanderlust_ team. We started off with many ideas and features that were seemed
+very appealing. However, we eventually decided that they were not suitable due to time constraints. In the first iteration, time and effort were placed
+in brainstorming and coming up with a concrete and feasible product. Throughout the weeks, we made consistent efforts to study the source code, modify existing features and
+create new features for _wanderlust_. There were high standards placed amongst us in ensuring that all of us put in time and effort in building _wanderlust).
```
Comment (lyeyixian) [inline docs/DeveloperGuide.md:1389]: Typo, you missed a "_" at the end.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1164,4 +1338,54 @@ testers are expected to do more *exploratory* testing.
...
+Rigorous testing, trial and error were fundamental in the process so that we can deliver _wanderlust_ within the timeline. Whilst _wanderlust_ may not be
+the perfect travel planner application, this journey was a memorable one filled with teamwork and perseverance, and we are all pround of the end product.
```
Comment (lyeyixian) [inline docs/DeveloperGuide.md:1391]: Typo, "proud"

**Review by lyeyixian — APPROVED** [review]
LGTM. Quite a few typos

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (jiaweiteo) [reply docs/DeveloperGuide.md:1364]: Resolved

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (jiaweiteo) [reply docs/DeveloperGuide.md:1379]: Resolved

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (jiaweiteo) [reply docs/DeveloperGuide.md:1383]: Resolved

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (jiaweiteo) [reply docs/DeveloperGuide.md:1389]: Resolved

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (jiaweiteo) [reply docs/DeveloperGuide.md:1391]: Resolved.

**Pull request merged**